### PR TITLE
Fix sitemap generation.

### PR DESCRIPTION
**Pull request by kabalin**
Regression caused changes in region controller (6eff83035, 65c875ddd).

### CODE REVIEW

**Comment by aeifn** [comment]
Кстати, чтобы повторно не пушить из-за линта, предусмотрен хук
https://github.com/PastVu/pastvu/blob/0b779a0be6762d8a5735c10f20eb6b4f741e0fc1/package.json#L38-L48

File: controllers/region.js
Comment (kabalin) [inline controllers/region.js:295]: Не очень мне нравится mixed return type в этой функции (promise|array), может быть вынести случай с fromDb в отдельную функцию?

File: controllers/region.js
Comment (klimashkin) [inline controllers/region.js:293]: fromDb is still optional, right?

File: controllers/photo.js
Comment (klimashkin) [inline controllers/photo.js:407]: Why do we need this changes? What are we fixing here?
`this.call()` sets the context, which might be used eventually

File: sitemap.js
Comment (klimashkin) [inline sitemap.js:38]: `1s` is not immediate. If we want 1s delay, why then we specify `immediate` flag?

**Review by klimashkin — CHANGES_REQUESTED** [review]

File: controllers/region.js
Comment (klimashkin) [reply controllers/region.js:295]: Make it async then

File: controllers/photo.js
Comment (kabalin) [reply controllers/photo.js:407]: We need this to make https://github.com/PastVu/pastvu/blob/master/sitemap.js#L213 line work correctly. getObjRegionList is used only in two places, in sitemap and photo controller. It does not look critical in this case to set context, but if you prefer I can rewrite its use in sitemap.

File: sitemap.js
Comment (kabalin) [reply sitemap.js:38]: May be it worth to comment why 4 is regarded as immediate 😃 (really, does not matter as long as it runs very soon). Other option would be to invoke `run` immediately, and set timeout to whatever the next sheduled run should be. This will be refactored anyway when we have job scheduler.

File: controllers/region.js
Comment (kabalin) [reply controllers/region.js:293]: Yes, correct (I had to check JSDoc, I took it as array first).

**Comment by kabalin** [comment]
Thanks @klimashkin for spending time reviewing it, I responded to some comments and will get follow up changes ready.

File: controllers/region.js
Comment (kabalin) [reply controllers/region.js:293]: BTW Correct syntax for destructing params https://jsdoc.app/tags-param.html#parameters-with-properties Anyway, leaving this as is for now (we may add JSDoc liniting one day).

File: sitemap.js
Comment (kabalin) [reply sitemap.js]: Reverted this one.

File: controllers/region.js
Comment (kabalin) [inline controllers/region.js:295]: ok, but this now cause a unnecessary chain of async/await in sitemap.js. May be it is actually better to split this function into two given we have two destinctive usecases and deprecate fromDb param?

**Comment by kabalin** [comment]
Just FYI, I verified the content of generated xml file, all looks good (no JS artefacts).

**Comment by kabalin** [comment]
Don't merge yet, some JS aretefact gets into xml.

**Comment by aeifn** [comment]
> Don't merge yet, some JS aretefact gets into xml.

I will not merge non-trivial PR without approval of @klimashkin.

**Comment by kabalin** [comment]
Ended up keeping region and photo controller code unchanged (apart of export statement). No async added, gets complicated and prone to errors in actual xml (at least without test infrastructure I would keep sitemap changes to minimum). Now works as expected.

**Review by klimashkin — APPROVED** [review]

File: controllers/region.js
Comment (klimashkin) [reply controllers/region.js]: You are right, feel free to change that. When these funtion was written, jsdoc didn't know about destructuring yet.

**Pull request merged**
